pipelines/toast_so_sim.py

```diff
@@ -1458,19 +1470,80 @@ def simulate_sky_signal(args, comm, data, schedules, subnpix, localsm):
     # Convolve a signal TOD from PySM
     if comm.comm_world.rank == 0:
         print("Simulating sky signal with PySM", flush=args.flush)
+
+    map_dist = (
+        None if comm is None else pysm.MapDistribution(nside=args.side, mpi_comm=comm)
+    )
+    pysm_component_objects = []
+    pysm_model = []
+    for model_tag in args.input_pysm_model.split(","):
+
+        if not model_tag.startswith("SO"):
+            pysm_model.append(model_tag)
+        else:
+
+            if model_tag == "SO_x1_cib":
+                pysm_component_objects.append(
+                    so_pysm_models.WebSkyCIB(
+                        websky_version="0.3",
+                        interpolation_kind="linear",
+                        nside=args.nside,
+                        map_dist=map_dist,
+                    )
+                )
+            elif model_tag == "SO_x1_ksz":
+                pysm_component_objects.append(
+                    so_pysm_models.WebSkySZ(
+                        version="0.3",
+                        nside=args.nside,
+                        map_dist=map_dist,
+                        sz_type="kinetic",
+                    )
+                )
+            elif model_tag == "SO_x1_tsz":
+                pysm_component_objects.append(
+                    so_pysm_models.WebSkySZ(
+                        version="0.3",
+                        nside=args.nside,
+                        map_dist=map_dist,
+                        sz_type="thermal",
+                    )
+                )
+            elif model_tag.startswith("SO_x1_cmb"):
+                lensed = model_tag.endswith("_lensed")
+                pysm_component_objects.append(
+                    so_pysm_models.WebSkyCMBMap(
+                        websky_version="0.3",
+                        lensed=lensed,
+                        seed=1,
+                        nside=args.nside,
+                        map_dist=map_dist,
+                    )
+                )
+            else:
+                pysm_component_objects.append(
+                    so_pysm_models.get_so_models(
+                        model_tag, args.nside, map_dist=map_dist
+                    )
+                )
+
+
     start = MPI.Wtime()
     signalname = "signal"
     op_sim_pysm = ttm.OpSimPySM(
         comm=comm.comm_rank,
         out=signalname,
-        pysm_model=args.input_pysm_model,
+        pysm_model=pysm_model,
+        pysm_component_objects=pysm_component_objects,
         focalplanes=[s[3] for s in schedules],
         nside=args.nside,
         subnpix=subnpix,
         localsm=localsm,
         apply_beam=args.apply_beam,
-        coord=args.coord,
+        coord="G", # setting G doesn't perform any rotation
+        map_dist=map_dist,
     )
+    assert args.coord == "Q", "Input SO models are always in Equatorial coordinates"
```

Comment:
notice we can only run in Equatorial for now, `so_pysm_models` defaults to Equatorial as well, so all input templates for PySM are in Equatorial, PySM doesn't perform any rotation (we set "G" for output and PySM defaults to input "G" templates as well).
If this is a problem please let me know and we'll think a better solution.